### PR TITLE
Implement HarmonicMechanicalSimulation results

### PR DESCRIPTION
**Pull request by PProfizi**
Implement result extraction APIs for HarmonicMechanicalSimulation.

# HarmonicMechanicalSimulation methods

(checked: tested, italic: implemented)

- [x] _displacement_
- [x] _velocity_
- [x] _acceleration_
- [x] _stress, stress_elemental, stress_nodal_
- [x] _stress_principal, stress_principal_nodal, stress_principal_elemental__
- [x] _stress_eqv_von_mises, stress_eqv_von_mises_elemental, stress_eqv_von_mises_nodal_
- [x] _elastic_strain, elastic_strain_nodal, elastic_strain_elemental_
- [x] _elastic_strain_principal, elastic_strain_principal_nodal, elastic_strain_principal_elemental_
- [ ] _elastic_strain_eqv_von_mises, elastic_strain_eqv_von_mises_elemental, elastic_strain_eqv_von_mises_nodal_
- [ ] elastic_strain_energy_density
- [x] _reaction_force_
- [x] _element_nodal_forces, element_nodal_forces_nodal, element_nodal_forces_elemental_
- [ ] _nodal_force_
- [ ] _nodal_moment_
- [ ] _thickness_
- [ ] _element_centroids_
- [ ] _element_orientations_
- [ ] _elemental_mass_
- [x] _elemental_volume_

### CODE REVIEW

File: src/ansys/dpf/post/harmonic_mechanical_simulation.py [hunk 1]
```diff
@@ -0,0 +1,2624 @@
...
+            all_sets:
+                Whether to get results for all sets.
+            load_steps:
+                Load steps to get results for.
+            named_selections:
+                Named selection or list of named selections to get results for.
+            selection:
+                Selection to get results for.
+                A Selection defines both spatial and time-like criteria for filtering.
+            location:
```
Comment (cbellot000) [inline src/ansys/dpf/post/harmonic_mechanical_simulation.py:556]: let's remove the string and add a ref to the locations doc

File: src/ansys/dpf/post/harmonic_mechanical_simulation.py [hunk 2]
```diff
@@ -0,0 +1,2624 @@
...
+                Location to extract results at. Available locations are listed in `locations`
+                and are: "Nodal", "Elemental", and "ElementalNodal". The default "ElementalNodal"
+                gives results with a value for every node at each element. "Elemental" gives results
+                with one value for each element. "Nodal" gives results with one value for each node.
+
+        Returns
+        -------
+            Returns a :class:`ansys.dpf.post.data_object.DataObject` instance.
```
Comment (cbellot000) [inline src/ansys/dpf/post/harmonic_mechanical_simulation.py:564]: P0: rename to dataframe

File: src/ansys/dpf/post/harmonic_mechanical_simulation.py [hunk 3]
```diff
@@ -0,0 +1,2624 @@
...
+            components:
+                Components to get results for. Available components are "X", "Y", "Z", "XX", "XY",
+                "XZ", and their respective equivalents 1, 2, 3, 4, 5, 6.
+            set_ids:
+                Sets to get results for.
+                A set is defined as a unique combination of {time, load step, sub-step}.
+            all_sets:
+                Whether to get results for all sets.
+            load_steps:
+                Load steps to get results for.
```
Comment (cbellot000) [inline src/ansys/dpf/post/harmonic_mechanical_simulation.py:690]: We can maybe give examples (especially for the tuple)

**Review by cbellot000 — APPROVED** [review]

File: src/ansys/dpf/post/harmonic_mechanical_simulation.py (hunk 2 above)
Comment (PProfizi) [reply src/ansys/dpf/post/harmonic_mechanical_simulation.py:564]: In next PR

**Pull request merged**